dependencies.yaml

```diff
@@ -108,28 +142,22 @@ dependencies:
             packages:
               - *gcc_aarch64
               - *sysroot_aarch64
-      - output_types: requirements
+      - output_types: [requirements, pyproject]
         matrices:
-          - matrix:
-              cuda: "11.8"
-            packages:
-              - "--extra-index-url=https://pypi.nvidia.com"
-              - cudf-cu11==23.6.*
-          - matrix:
-              cuda: "11.5"
-            packages:
-              - "--extra-index-url=https://pypi.nvidia.com"
-              - cudf-cu11==23.6.*
-          - matrix:
-              cuda: "11.4"
-            packages:
-              - "--extra-index-url=https://pypi.nvidia.com"
-              - cudf-cu11==23.6.*
-          - matrix:
-              cuda: "11.2"
-            packages:
-              - "--extra-index-url=https://pypi.nvidia.com"
-              - cudf-cu11==23.6.*
+          - matrix: {cuda: "11.8"}
```

Comment:
Could we retain the "longer" form of YAML with more newlines/etc.? We don't use the shorter inline syntax like `{matrix: {cuda: "11.5}, ...}` in any other `dependencies.yaml` files afaik.

Reply:
If we do, IMO this file gets much more difficult to parse (and will only get worse the over time). Much prefer the shorter syntax here.

Reply:
Paul has explained to me that there are other (WIP) reasons to prefer this syntax.

Reply:
I tend to agree with Bradley here, I find the longer-form syntax more readable despite being more verbose. The lines below where the packages are also collapsed into the same line aren't my favorite, for instance. It's sort of like reading JSON where any pretty printer will split the fields into multiple lines.

I won't personally block on this change though since I don't read cuspatial code enough.